statsd/statsd.go

```diff
@@ -271,11 +286,25 @@ func newWithWriter(w statsdWriter, o *Options, writerName string) (*Client, erro
 		telemetryTags: []string{clientTelemetryTag, clientVersionTelemetryTag, "client_transport:" + writerName},
 	}
 
-	// Inject DD_ENTITY_ID as a constant tag if found
-	entityID := os.Getenv(entityIDEnvName)
-	if entityID != "" {
-		entityTag := fmt.Sprintf("%s:%s", entityIDTagName, entityID)
-		c.Tags = append(c.Tags, entityTag)
+	// Inject all tags contained in DD_TAGS. Each tag is of form "name:value" or simply "name".
+	for _, tag := range strings.Split(os.Getenv(ddTagsEnvName), ",") {
+		if tag == "" {
+			continue
+		}
+		parts := strings.Split(tag, ":")
+		if len(parts) == 1 {
+			c.Tags = append(c.Tags, parts[0])
+		}
+		if len(parts) == 2 {
+			c.Tags = append(c.Tags, fmt.Sprintf("%s:%s", parts[0], parts[1]))
+		}
```

Comment:
Why split ? What is the value of a tag contains `:` ?

Reply:
Good call, we do support `:` in tag values.

I've updated the logic and test cases accordingly.